falcon/request.py

```diff
@@ -475,6 +480,59 @@ def headers(self):
     def params(self):
         return self._params
 
+    @property
+    def param(self):
+        if self._param is None:
+            self._param = helpers.ParamProxy(self, self.params)
+
+        return self._param
+
+    @property
+    def raw_body(self):
+        # NOTE(necaris): This will only store the raw body data if
+        # `self.options.store_raw_body` is True; otherwise it will consume
+        # the body data stream and discard data after returning
+        if self._raw_body is not None:
+            raw_body = self._raw_body
+        else:
+            # NOTE(kgriffs): This assumes self.stream has been patched
+            # above in the case of wsgiref, so that self.content_length
+            # is not needed. Normally we just avoid accessing
+            # self.content_length, because it is a little expensive
+            # to call. We could cache self.content_length, but the
+            # overhead to do that won't usually be helpful, since
+            # content length will only ever be read once per
+            # request in most cases.
+            raw_body = self.stream.read()
+
+            if self.options.store_raw_body:
+                self._raw_body = raw_body
+
+        return raw_body
+
+    @property
+    def form_params(self):
+        if self._form_params is None:
+            # PERF(kgriffs): Technically, we should spend a few more
+            # cycles and parse the content type for real, but
+            # these heuristics will work virtually all the time.
+            content_type = self.content_type or ''
```

Comment:
Explicitly checking `is None` (as was done before this change) is slightly more performant.